buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/publish/PublishingPlugin.groovy

```diff
@@ -121,11 +131,40 @@ class PublishingPlugin implements Plugin<Project> {
                             publisher.decorate(sub, it)
                         }
                     }
+
                     publishAllToLocal.dependsOn "$sub.path:publishMavenAarPublicationToMavenLocal"
                     publishAllToBuildDir.dependsOn "$sub.path:publishMavenAarPublicationToBuildDirRepository"
 
+                    // Update the changed and all sets each time an artifact is published. This is
+                    // used to build the JSON file of affected targets.
+                    sub.tasks.getByName("publishMavenAarPublicationToBuildDirRepository").doLast {
```

Comment:
Why does this have to be part of this task? Don't we have all the required info at gradle startup/configuration phase?

Reply:
I suppose it doesn't. I was thinking of adding each as it finished being built, but I suppose there's no real reason for that.